components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -456,100 +459,108 @@ func InjectCertConfig(notebook *nbv1.Notebook, configMapName string) error {
+
+						// Split the imageSelection to imagestream and tag
+						imageSelected := strings.Split(imageSelection, ":")
+						if len(imageSelected) != 2 {
+							log.Error(nil, "Invalid image selection format")
+							return fmt.Errorf("invalid image selection format")
+						}
+
+						// Specify the namespaces to search in
+						namespaces := []string{"opendatahub", "redhat-ods-applications"}
+						imagestreamFound := false
+						for _, namespace := range namespaces {
+							// List imagestreams in the specified namespace
+							imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
+							if err != nil {
+								log.Error(err, "Cannot list imagestreams", "namespace", namespace)
+								continue
+							}
+
+							// Iterate through the imagestreams to find matches
+							for _, item := range imagestreams.Items {
+								metadata := item.Object["metadata"].(map[string]interface{})
+								name := metadata["name"].(string)
+
+								// Match with the ImageStream name
+								if name == imageSelected[0] {
+									status := item.Object["status"].(map[string]interface{})
+
+									// Match to the corresponding tag of the image
+									tags := status["tags"].([]interface{})
+									for _, t := range tags {
+										tagMap := t.(map[string]interface{})
+										tagName := tagMap["tag"].(string)
+										if tagName == imageSelected[1] {
+											items := tagMap["items"].([]interface{})
+											if len(items) > 0 {
+												// Sort items by creationTimestamp to get the most recent one
+												sort.Slice(items, func(i, j int) bool {
+													iTime := items[i].(map[string]interface{})["created"].(string)
+													jTime := items[j].(map[string]interface{})["created"].(string)
+													return iTime > jTime // Lexicographical comparison of RFC3339 timestamps
+												})
+												imageHash := items[0].(map[string]interface{})["dockerImageReference"].(string)
+												// Update the Containers[i].Image value
+												notebook.Spec.Template.Spec.Containers[i].Image = imageHash
+												// Update the JUPYTER_IMAGE environment variable with the image selection for example "jupyter-datascience-notebook:2023.2"
+												for i, envVar := range container.Env {
+													if envVar.Name == "JUPYTER_IMAGE" {
+														container.Env[i].Value = imageSelection
+														break
+													}
+												}
+												imagestreamFound = true
+												break
+											}
+										}
+									}
+								}
+							}
+							if imagestreamFound {
+								break
+							}
+						}
+						if !imagestreamFound {
+							log.Info("Imagestream not found in any of the specified namespaces", "imageSelected", imageSelected[0], "tag", imageSelected[1])
+						}
+					}
+				}
+			}
+			if !containerFound {
+				log.Info("No container found matching the notebook name", "notebookName", notebook.Name)
```

Comment:
Is this info or rather error/warning message? This situation is unexpected and shouldn't happen in standard scenario, correct? If so, I suppose error is better? :thinking:

Reply:
done